www/pages/routes-and-stops/routes-and-stops-controller.js

```diff
@@ -8,43 +8,58 @@ angular.module('pvta.controllers').controller('RoutesAndStopsController', functi
   /*
    * Get all the routes and stops
    */
+
+   $scope.redirectRoute = function(routeId){
+    $state.go('app.route', {routeId: routeId});
+  }
+
+  $scope.redirectStop = function(stopId){
+    $state.go('app.stop', {stopId: stopId});
+  }
+
   function getRoutesAndStops () {
     $ionicLoading.show();
     // RouteForage returns a promise, resolve it.
     RouteForage.get().then(function (routes) {
       RouteForage.save(routes);
-      $scope.routes = stripDetails(routes);
+      getFavoriteRoutes(routes);
       redraw();
     });
     /*
     * Nested function for removing stuff we don't need
     * from each route; this makes searching easier!
     */
-    function stripDetails (routeList) {
-      return _.map(routeList, function (route) {
-        return _.pick(route, 'RouteId', 'RouteAbbreviation', 'GoogleDescription', 'ShortName', 'Color');
-      });
-    }
     // Remember, StopsForage returns a Promise.
     // Must resolve it.
     StopsForage.get().then(function (stops) {
       stops = StopsForage.uniq(stops);
-      $scope.stops = prepareStops(stops);
       StopsForage.save(stops);
-      $ionicLoading.hide();
+      getFavoriteStops(stops);
       redraw();
     });
 
+    $ionicLoading.hide();
+
     /* Similar to prepareRoutes, we only
      * keep the details about each stop that are useful
      * to us for displaying them.  It makes searching easier.
      */
-    function prepareStops (list) {
-      return _.map(list, function (stop) {
-        return _.pick(stop, 'StopId', 'Description');
-      });
-    }
   }
+
+  function stripDetails (routeList) {
```

Comment:
We have a function called `prepareStops`, why not rename this to `prepareRoutes?`